src/App.tsx

```diff
@@ -1,42 +1,176 @@
+import { ColorTheme, DefaultTheme } from "@/theme/colors";
+import { Space } from "@/theme/layouts";
+import { Typography } from "@/theme/typography";
+import AddProfileScreen from "@/pages/default/profiles/AddProfile";
+import AddCredentialDetailScreen from "./pages/default/credentials/AddCredentialDetail";
+import AddCredentialOptionsScreen from "./pages/default/credentials/AddCredentialOptions";
 
-export default function App() {
-  useEffect(() => {
-    const startupTasks = async () => {
-      await DwnService.initExpoLevelDwn();
-      await bootstrapIdentityAgent(
-        "passphrase",
-        "Personal",
-        DwnService.getDwn()
-      );
-    };
-    void startupTasks();
-  }, []);
+const App = () => {
+  const Stack = createNativeStackNavigator();
+  const initialRoute = userProfiles.peek().length ? "Default" : "Onboarding";
 
   return (
-    <NavigationContainer theme={DarkTheme} linking={linking}>
-      <StatusBar style="light" />
-      <PaperProvider theme={theme}>
-        <AppNavigator />
-      </PaperProvider>
+    <NavigationContainer linking={DeepLinkConfig} theme={DefaultTheme}>
+      <Stack.Navigator
+        screenOptions={{ headerShown: false }}
+        initialRouteName={initialRoute}
+      >
+        {/* Onboarding */}
+        <Stack.Screen name="Onboarding" component={WelcomeScreen} />
+        <Stack.Screen name="Import" component={ImportScreen} />
+        <Stack.Screen name="Create" component={CreateScreen} />
+        <Stack.Screen name="Auth" component={AuthScreen} />
+
+        {/* Default */}
+        <Stack.Screen name="Default" component={TabNavigation} />
+        <Stack.Group
+          screenOptions={{
+            headerShown: true,
+            headerTitle: () => false,
+            headerShadowVisible: false,
+            headerBackTitleVisible: false,
+            // headerBackImageSource: TODO: add arrow image source here
+          }}
+        >
+          <Stack.Screen name="ProfileDetail" component={ProfileDetailScreen} />
+          <Stack.Screen
+            name="CredentialDetail"
+            component={CredentialDetailScreen}
+          />
+          <Stack.Screen name="AddProfile" component={AddProfileScreen} />
+          <Stack.Screen
+            name="AddCredentials"
+            component={AddCredentialsScreen}
+          />
+          <Stack.Screen
+            name="AddCredentialDetail"
+            component={AddCredentialDetailScreen}
+          />
+          <Stack.Screen
+            name="AddCredentialOptions"
+            component={AddCredentialOptionsScreen}
+          />
+          <Stack.Screen
+            name="ConnectionDetail"
+            component={ConnectionDetailScreen}
+          />
+          <Stack.Screen
+            name="ReviewConnection"
+            component={ReviewConnectionScreen}
+          />
+        </Stack.Group>
+      </Stack.Navigator>
     </NavigationContainer>
   );
-}
+};
+
+export default App;
+
+const TabNavigation = () => {
```

Comment:
we should preserve the separation of concerns by keeping the navigation in separate folders and files but I can fix that in a follow up PR